### PR TITLE
amap: init at 5.4

### PR DESCRIPTION
**Pull request by octodi**
## Description of changes

Added amap (AMAP stands for Application MAPper. It is a next-generation scanning tool for pentesters.)

https://gitlab.com/kalilinux/packages/amap/

Related to https://github.com/NixOS/nixpkgs/issues/81418

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by donovanglover — CHANGES_REQUESTED** [review]
Is there a reason to use `amap`? Upstream recommends to use `nmap` instead.

Other distributions like Gentoo have [removed](https://repology.org/project/amap-pentesting/history) amap from their repositories.

File: pkgs/by-name/am/amap/package.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  buildInputs = with pkgs; [
+    libuecc
+  ];
+
+  meta = with lib; {
+    description = "A next generation scanning tool";
+    homepage = "https://gitlab.com/kalilinux/packages/amap";
+    changelog = "https://gitlab.com/packages/amap/-/blob/${src.rev}/CHANGES";
+    license = licenses.gpl2Only;
```
[the remark this answers is not in the document]
Comment (octodi) [reply pkgs/by-name/am/amap/package.nix:36]: In it's source repo it clearly mentions GPL2

[the remark this answers is not in the document]
Comment (octodi) [reply pkgs/by-name/am/amap/package.nix:36]: `
			LICENCE FOR AMAP (all version)
                     by van Hauser <vh@thc.org>


1. This software comes with no warrenty or promised features. If it works
for you - fine. It just comes "AS-IS", which means as a bunch of bits and bytes.

2. Anyone may use this software and pass it on to other persons or companies
as long as it is not charged for! (except for a small transfer/medium fee)

3. This tool may *NOT* be used for illegal purpose. Please check the law
which affects your doing. I will have got no liability for any damage etc.
done with this tool legally or illegaly.

4. If this tool is used while providing a commercial service (e.g. as part
of a penetration test) the report has to state the tools name and version,
and additionally the authors (van Hauser and Dj RevMoon) and the distribution
homepage (http://www.thc.org).

5. If this tool is used within a commercial tool (being called out of such a
tool or being incorporated), the report generated has to state the tools name
and version, and additionally the authors (van Hauser and Dj RevMoon) and the
distribution homepage (http://www.thc.org). A tool is "commercial" if it either
costs money to purchase it, has a license fee, and/or has costs for upgrades.
Additionally, a commercial version or license etc. must be made available to
the author free of charge.

6. In all other respects the GPL 2.0 applies
`

Yup it uses custom license with [GPL2](https://gitlab.com/kalilinux/packages/amap/-/blob/kali/master/LICENSE.GNU), so should I change it?

**Comment by octodi** [comment]
> Is there a reason to use `amap`? Upstream recommends to use `nmap` instead.
> 
> Other distributions like Gentoo have [removed](https://repology.org/project/amap-pentesting/history) amap from their repositories.

I was adding it because it was in [kali](https://www.kali.org/tools/amap/) repo, blackarch also doesn't include it

File: pkgs/by-name/am/amap/package.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (donovanglover) [reply pkgs/by-name/am/amap/package.nix:36]: I think so, although I don't have experience with packaging custom licensed software. Looks like `nmap` also has a custom license but nixpkgs has it as GPLv2. Overall licenses seem [inconsistent](https://repology.org/project/nmap/information) across distributions.

**Comment by donovanglover** [comment]
I don't think we should add `amap` to nixpkgs personally. As far as I'm aware, there isn't a reason to use `amap` over `nmap`, which is already in nixpkgs.

**Comment by octodi** [comment]
> I don't think we should add `amap` to nixpkgs personally. As far as I'm aware, there isn't a reason to use `amap` over `nmap`, which is already in nixpkgs.

Okay

**Pull request closed without merging**